### PR TITLE
Feature flag ctags metadata generation

### PR DESCRIPTION
**Pull request by kzh**
This PR adds a flag `disable_ctags` for disabling ctags during indexing. It also stores a boolean indicating whether an index contains ctags metadata.

### CODE REVIEW

File: api.go [hunk 1]
```diff
@@ -217,6 +217,8 @@ type Repository struct {
 	// IndexOptions is a hash of the options used to create the index for the
 	// repo.
 	IndexOptions string
+
+	Symbols bool
```
Comment (keegancsmith) [inline api.go]: Add a doc string please. Maybe a better name is `HasSymbols`

File: cmd/flags.go [hunk 2]
```diff
@@ -38,14 +38,15 @@ func (f *largeFilesFlag) Set(value string) error {
...
-	version     = flag.Bool("version", false, "Print version number")
-	ctags       = flag.Bool("require_ctags", false, "If set, ctags calls must succeed.")
-	largeFiles  = largeFilesFlag{}
+	sizeMax      = flag.Int("file_limit", 128*1024, "maximum file size")
+	trigramMax   = flag.Int("max_trigram_count", 20000, "maximum number of trigrams per document")
+	shardLimit   = flag.Int("shard_limit", 100<<20, "maximum corpus size for a shard")
+	parallelism  = flag.Int("parallelism", 4, "maximum number of parallel indexing processes.")
+	indexDir     = flag.String("index", build.DefaultDir, "directory for search indices")
+	version      = flag.Bool("version", false, "Print version number")
+	disableCTags = flag.Bool("disable_ctags", false, "Generate ctags metadata")
```
Comment (keegancsmith) [inline cmd/flags.go]: incorrect explanation of the flag. Suggestion below.
```suggestion
	disableCTags = flag.Bool("disable_ctags", false, "If set, ctags will not be called.")
```

File: cmd/zoekt-sourcegraph-indexserver/main.go [hunk 3]
```diff
@@ -141,10 +141,16 @@ type IndexOptions struct {
...
 	LargeFiles []string
+
+	// Symbols is a boolean that indicates whether to generate ctags metadata or not
+	Symbols bool
 }
 
 func (o *IndexOptions) toArgs() []string {
-	args := make([]string, 0, len(o.LargeFiles)*2)
+	args := make([]string, 0, len(o.LargeFiles)*2+1)
+	if !o.Symbols {
```
Comment (keegancsmith) [inline cmd/zoekt-sourcegraph-indexserver/main.go]: if its true can we set `-require_ctags`

File: toc.go [hunk 4]
```diff
@@ -39,7 +39,8 @@ const IndexFormatVersion = 16
 // 6: Include '#' into the LineFragment template
 // 7: Record skip reasons in the index.
 // 8: Record source path in the index.
-const FeatureVersion = 8
+// 9: Store ctags metadata
+const FeatureVersion = 9
```
Comment (keegancsmith) [inline toc.go:43]: aah good catch. Last minute thought, but any chance we can store the ctags data after everything else is written such that reading the data in older zoekt just works? Then we won't require a reindex.

File: write.go [hunk 5]
```diff
@@ -156,6 +156,7 @@ func (b *IndexBuilder) Write(out io.Writer) error {
 	}, &toc.metaData, w); err != nil {
 		return err
 	}
+	b.repo.Symbols = len(b.runeDocSections) > 0
```
Comment (keegancsmith) [inline write.go]: if this is how we set the value, why do we need to write it? For the `-incremental` support we can just rely on HashOptions?

**Review by keegancsmith — APPROVED** [review]
Some changes inline

File: toc.go (hunk 4 above)
Comment (kzh) [reply toc.go:43]: This would be a tool of its own for migrating between Zoekt index versions.

Comment (keegancsmith) [reply toc.go:43]: aah ok, nevermind then :)

**Review by keegancsmith — COMMENTED** [review]
LGTM

**Pull request merged**